src/yunohost/backup.py

```diff
@@ -969,24 +972,25 @@ def unknown_error(part):
                 continue
 
             # Otherwise, attempt to find it (or them?) in the archive
-            hook_paths = '{:s}/hooks/restore/*-{:s}'.format(self.work_dir, system_part)
-            hook_paths = glob(hook_paths)
 
             # If we didn't find it, we ain't gonna be able to restore it
-            if len(hook_paths) == 0:
+            if system_part not in self.info['system'] or\
+                    'paths' not in self.info['system'][system_part] or\
+                    len(self.info['system'][system_part]['paths']) == 0:
                 logger.exception(m18n.n('restore_hook_unavailable', part=system_part))
                 self.targets.set_result("system", system_part, "Skipped")
                 continue
 
+            hook_paths = self.info['system'][system_part]['paths']
+            hook_paths = [ 'hooks/restore/%s' % os.path.basename(p) for p in hook_paths ]
+
             # Otherwise, add it from the archive to the system
             # FIXME: Refactor hook_add and use it instead
-            custom_restore_hook_folder = os.path.join(CUSTOM_HOOK_FOLDER, 'restore')
-            filesystem.mkdir(custom_restore_hook_folder, 755, True)
             for hook_path in hook_paths:
                 logger.debug("Adding restoration script '%s' to the system "
                              "from the backup archive '%s'", hook_path,
                              self.archive_path)
-            shutil.copy(hook_path, custom_restore_hook_folder)
+                self.method.copy(hook_path, custom_restore_hook_folder)
```

Comment:
I'm trying this on a dev instance ... 
- I created an archive with a custom hook (or rather : I removed the default restore hook for conf_cron)
- But then when I get to this line, I have 
   - `hook_path = "/hooks/restore/32-conf_cron"`
   - `custom_restore_hook_folder = /etc/yunohost/hooks.d/restore`
- which ends up with the "real" file created/restored being : `/etc/yunohost/hooks.d/restore/hooks/restore/32-conf_cron` ... instead of `/etc/yunohost/hooks.d/restore/32-conf_cron` 
- annnnnd I don't know how to properly fix this :| Maybe there's some trick to do somewhere to tweak the destination path somehow ...

But apart from this, that fix sounds epic o.O

Reply:
Thank you for the review, https://github.com/YunoHost/yunohost/pull/927/commits/cb5ca42df01ba61067d6d77001b596597bd05f3c should fix this issue :-)